database/migrations/000006_status_delete_procedure.up.sql

```diff
@@ -0,0 +1,116 @@
+-- Copyright 2023 Stacklok, Inc
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+-- Start to make sure the function and trigger are either both added or none
+BEGIN;
+
+-- Define the trigger function (if it doesn't already exist)
+CREATE OR REPLACE FUNCTION update_profile_status_on_delete() RETURNS TRIGGER AS $$
+DECLARE
+    v_status eval_status_types;
+BEGIN
+    SELECT CASE
+           WHEN EXISTS (
+                SELECT 1 FROM rule_details_eval rde
+                INNER JOIN rule_evaluations res ON res.id = rde.rule_eval_id
+                WHERE profile_id = OLD.profile_id AND status = 'error'
+           ) THEN 'error'
+           WHEN EXISTS (
+               SELECT 1 FROM rule_details_eval rde
+               INNER JOIN rule_evaluations res ON res.id = rde.rule_eval_id
+               WHERE profile_id = OLD.profile_id AND status = 'failure'
+           ) THEN 'failure'
+           WHEN NOT EXISTS (
+               SELECT 1 FROM rule_details_eval rde
+                                 INNER JOIN rule_evaluations res ON res.id = rde.rule_eval_id
+               WHERE profile_id = OLD.profile_id
+           ) THEN 'pending'
+           WHEN NOT EXISTS (
+               SELECT 1 FROM rule_details_eval rde
+               INNER JOIN rule_evaluations res ON res.id = rde.rule_eval_id
+               WHERE profile_id = OLD.profile_id AND status != 'skipped'
+           ) THEN 'skipped'
+           WHEN NOT EXISTS (
+               SELECT 1 FROM rule_details_eval rde
+                                 INNER JOIN rule_evaluations res ON res.id = rde.rule_eval_id
+               WHERE profile_id = OLD.profile_id AND status NOT IN ('success', 'skipped')
+           ) THEN 'success'
+           ELSE (
+               'error' -- This should never happen, if yes, make it visible
+               )
+   END INTO v_status;
+
+   UPDATE profile_status SET profile_status = v_status, last_updated = NOW()
+   WHERE profile_id = OLD.profile_id;
+
+   RETURN NULL;
+END;
+$$ LANGUAGE plpgsql;
+
+-- Create the trigger
+CREATE TRIGGER update_profile_status_after_delete
+    AFTER DELETE ON rule_evaluations
+    FOR EACH ROW
+    EXECUTE FUNCTION update_profile_status_on_delete();
+
+-- Update overall profile status if a rule evaluation status is updated
+-- error takes precedence over failure, failure takes precedence over success
+CREATE OR REPLACE FUNCTION update_profile_status() RETURNS TRIGGER AS $$
```

Comment:
Can you change the comment to mention what actually changed? It's tricky to figure out since we don't have a diff

Reply:
sure. I just realised I should have included the old function in the down migration for completeness, too